### DISCUSSION TITLE
Creating item including translations through ItemsService don't show in App.

### DISCUSSION THREAD

**Discussion by keesvanbemmel** [directus/directus#9859] [Archive]
Hi all,

It seems that creating an item including translations don't show in the app if I create them through the ItemsService. Surely I'm missing something. This is my code:
```
const categoriesService = new ItemsService('categories', {
      schema,
    });

    const newCategory = {
      id: 1342,
      slug: 'test-kees',
      translations: [
        { categories_slug: 'test-kees', name: 'en-test-kees-name', language_code: 'en-US' },
        { categories_slug: 'test-kees', name: 'nl-test-kees-name', language_code: 'nl-NL' },
      ],
    };

    const newId = await categoriesService.upsertOne(newCategory);
    logger.info(`New category id: ${newId}`);
```

The Category item is created. The Category_Translation items are also created. They are links. However, nothing shows in the app translation wise. This is the warning my browser console gives me:
`The specified value "nextval('categories_translations_id_seq'::regclass)" cannot be parsed, or is out of range.`

See screenshots of the app for both the category list, category translation list and the specific category. I would've suspected the en-test-kees-name to show up there.

Any thoughts?

My second question: I'm upserting these categories from an external provider. Is there a way to upsert them correctly without knowing the translation id's at that time?

<img width="1432" alt="CleanShot 2021-11-16 at 19 12 08@2x" src="https://user-images.githubusercontent.com/4633876/142041864-0d2d6603-9423-48aa-b0f6-2ea7fa73d599.png">

<img width="1435" alt="CleanShot 2021-11-16 at 19 12 41@2x" src="https://user-images.githubusercontent.com/4633876/142041927-d370b3ca-919a-4945-a5ae-491dfc94692a.png">

<img width="1781" alt="CleanShot 2021-11-16 at 19 11 45@2x" src="https://user-images.githubusercontent.com/4633876/142041804-273f6d7f-8d4e-4df5-9a23-97dc6b1c3d69.png">

**Comment by rijkvanzanten** [comment]
Heya!

What is the `categories_slug` field in this case? Is that the m2o back to the parent categories collections?

For what it's worth, the application works the exact same way. When you create a new item with new translations, the following payload is sent:

```json
{
  "translations": [
    { "title": "Nederlandse Titel", "language_code": "nl-NL" },
    { "title": "English Title", "language_code": "en-US" }
  ]
}
```

This is then passed as-is to the ItemsService to be saved, using the same logic you're relying on in your custom code:

https://github.com/directus/directus/blob/21a053f19a9a017336b39fccfde6a04d8bcd41a1/api/src/controllers/items.ts#L22-L25

https://github.com/directus/directus/blob/21a053f19a9a017336b39fccfde6a04d8bcd41a1/api/src/controllers/items.ts#L33

The only thing I can think of that might screw with this is the connection from the junction table to the original categories, or the junction table to the languages table. Can you confirm that all field names that you're using match the exact field names configured in the relationship? and that the relationship is configured as a many to many from categories->translations->languages?
